qemu/tests/nic_opt.py

```diff
@@ -0,0 +1,149 @@
+import os
+import logging
+import time
+
+from virttest import error_context
+from virttest import utils_net
+from virttest import utils_netperf
+from virttest import utils_misc
+from virttest import utils_test
+from virttest import data_dir
+from avocado.utils import cpu
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    NIC option check test.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def test_netperf():
+        """
+        Netperf stress test for nic option.
+        """
+        netperf_client_link = os.path.join(data_dir.get_deps_dir("netperf"),
+                                           params.get("netperf_client_link"))
+        client_path = params.get("client_path")
+
+        n_client = utils_netperf.NetperfClient(
+            vm.get_address(), client_path,
+            netperf_source=netperf_client_link, client=params.get("shell_client"),
+            port=params.get("shell_port"), username=params.get("username"),
+            password=params.get("password"), prompt=params.get("shell_prompt"),
+            linesep=params.get("shell_linesep", "\n").encode().decode(
+                'unicode_escape'),
+            status_test_command=params.get("status_test_command", ""))
+
+        n_server = utils_netperf.NetperfServer(
+            utils_net.get_host_ip_address(params),
+            params.get("server_path", "/var/tmp"),
+            params.get("pkg_md5sum"),
+            os.path.join(data_dir.get_deps_dir("netperf"),
+                         params.get("netperf_server_link")),
+            password=params.get("hostpassword"))
+
+        try:
+            n_server.start()
+            # Run netperf with message size defined in range.
+            netperf_test_duration = int(params.get("netperf_test_duration", 180))
+            netperf_para_sess = params.get("netperf_para_sessions", "1")
+            test_protocols = params.get("test_protocols", "TCP_STREAM")
+            netperf_cmd_prefix = params.get("netperf_cmd_prefix", "")
+            netperf_output_unit = params.get("netperf_output_unit", " ")
+            netperf_package_sizes = params.get("netperf_sizes")
+            test_option = params.get("test_option", "")
+            test_option += " -l %s" % netperf_test_duration
+            if params.get("netperf_remote_cpu") == "yes":
```

Comment:
local  and remote should be a pair of conflict options, so it should be   in if ... else ... branch, right?

Reply:
Both "netperf_remote_cpu" and "netperf_local_cpu" are required by netperf.The test command is: netperf -H $ip -C -c
-t TCP_STREAM -l 180s.